### DISCUSSION TITLE
Is there anyway to get where message comes from?

### DISCUSSION THREAD

**Discussion by muhcdan** [nextgenhealthcare/connect#5863] [General]
I often look at the source in the dashboard to see the data but I can't see anywhere where it shows me from what IP did this message come from. 

Anyway go get that information?

TIA

**Comment by pacmano1** [comment]
Go to the message viewer, view a single message and it will show where it came from.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/f040734c-526b-4902-a74c-5d35279e7224)

**Reply by pacmano1** [reply]
@muhcdan Please starting using the Q&A discssion category rather than the general category - just so an answer can be marked as such.   The "general" category doesn't have that feature.

**Reply by muhcdan** [reply]
OK. Next time will post to Q&A. I don't see those. I'm running version 3.12.

**Reply by pacmano1** [reply]
What kind of listener is it?    And that screenshot is from 3.9.1, from a TCP listener, so 3.12 should have it.

Show a screenshot.

**Reply by muhcdan** [reply]
[this is a web service listener](https://user-images.githubusercontent.com/140529208/256326672-123bc80b-e358-40d9-9992-13b55cde5a99.png)

**Reply by pacmano1** [reply]
Yet another reason I don't use web service listeners or senders, lol.   Rather, http listeners and senders.    Man I hate that connector type.

**Reply by muhcdan** [reply]
You're right I do see them on tcp and http connectors. Thanks
